src/maphttp.c

```diff
@@ -484,6 +484,9 @@ int msHTTPExecuteRequests(httpRequestObj *pasReqInfo, int numRequests,
     return (MS_FAILURE);
   }
 
+  struct curl_slist **headersArray =
+      msSmallMalloc(numRequests * sizeof(struct curl_slist *));
```

Comment:
to make sure entries are null initialized:
```suggestion
      msSmallCalloc(numRequests, sizeof(struct curl_slist *));
```